src/composables/useLeafletButtonControl.ts

```diff
@@ -1,9 +1,17 @@
-import { ref } from 'vue'
-import { ButtonControl } from '@/models/leafletButtonControl'
+import {Ref, ref} from 'vue'
+import {ButtonControl, StateChangeType} from '@/models/leafletButtonControl'
+
+export function useLeafletButtonControl(
+  iconKey: string,
+  tooltipTitle: string,
+  stateChangeHook?: (
+    changedState: Ref<boolean>,
```

Comment:
Do we want a ref here? I think we want the current state that was changed at the time the hook was called

Reply:
In general I'm not sure if the callback provides a significant benefit over simply using a watcher on the ref. Both are triggered on every state change. The watcher might add a little overhead but the callback clutters the interface so it might be better to just remove the callback altogether.